### PR TITLE
Feat/swarinfo 542 user guides 1

### PR DESCRIPTION
**Pull request by stlaperriere**
First User Guide : blinky example

### CODE REVIEW

**Review by cquesnel — APPROVED** [review]

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 1]
```diff
@@ -1,11 +1,205 @@
...
+
+function create_exec(){
+    exec = executor.new(10, tick, ctx);
+    return exec;
+}
+```
+
+Copy this snippet in `blinky.bzz`. Save the file, and flash the code to the HiveBoard. The HiveBoard will reboot and you should see the MCU2 light blinking.
+
+You'll have noticed in the code that the `tick` function uses a logger. To view the output, use your favorite tool to open the `ttyUSB2` COM port. You should see the following output:
```
Comment (xgroleau) [inline docs/sections/user-guide/basic-examples/led-flash-buzz.md:154]: ttyUSB2 changes from computer to computer, udev rules and on the other peripherals plugged in the user pc. You could say that it `should be` ttyUSB2, but it's not an absolute truth.
You can know if it's ttyUSB2 with
`ls -l /dev/serial/by-id`

**Review by xgroleau — APPROVED** [review]

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 2]
```diff
@@ -1,11 +1,205 @@
...
+
+![Connecting and Powering a HiveBoard](img/connect-hiveboard.png)
+
+Next, let's flash the HiveMind firmware to the HiveBoard. Follow the steps written in [HiveMind's README file](https://github.com/SwarmUS/HiveMind/blob/master/README.md) to build and flash your board. **Follow the steps for the H7/HiveBoard**.
+
+> Take a close attention to the requirements listed in the README and make sure you use the correct versions of CMake, GCC, arm-gcc, and OpenOCD. 
+
+In HiveMind's README, you will find some steps to build using ROS. You need not follow these steps for now; the ROS build is only used in simulation. This how-to guide only shows how to compile and flash for the HiveBoard.
+
+Once you have successfully flashed the firmware to the board, you may go to the next step, where you will write, compile, and flash a Buzz script.
```
[the remark this answers is not in the document]
Comment (stlaperriere) [reply docs/sections/user-guide/basic-examples/led-flash-buzz.md:27]: done

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 3]
```diff
@@ -1,11 +1,205 @@
...
+function init() {
+  exec = create_exec()
+}
+
+function step() {
+    exec.step();
+}
+```
+
+The main file is used to select the Buzz script to be run, by `include`-ing the right file. Usually, the main file is not meant to be very long, as all the applicative logic should be in the included file.
```
[the remark this answers is not in the document]
Comment (stlaperriere) [reply docs/sections/user-guide/basic-examples/led-flash-buzz.md:52]: removed

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 4]
```diff
@@ -1,11 +1,205 @@
...
+
+Start by connecting the HiveBoard to your computer using a micro-USB cable. Next, plug in the power supply using the wall adapter. Turn on the power by setting the power swith to `CON`. You should see the HiveBoard light up. If the HiveBoard already has a firmware, it should output its logs via a COM port (usually `ttyUSB2`). You might want to check the output using your favorite tool.
+
+![Connecting and Powering a HiveBoard](img/connect-hiveboard.png)
+
+Next, let's flash the HiveMind firmware to the HiveBoard. Follow the steps written in [HiveMind's README file](https://github.com/SwarmUS/HiveMind/blob/master/README.md) to build and flash your board. **Follow the steps for the H7/HiveBoard**.
+
+> Take a close attention to the requirements listed in the README and make sure you use the correct versions of CMake, GCC, arm-gcc, and OpenOCD. 
+
+In HiveMind's README, you will find some steps to build using ROS. You need not follow these steps for now; the ROS build is only used in simulation. This how-to guide only shows how to compile and flash for the HiveBoard.
```
[the remark this answers is not in the document]
Comment (stlaperriere) [reply docs/sections/user-guide/basic-examples/led-flash-buzz.md:25]: just me trying to be fancy here

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 5]
```diff
@@ -1,11 +1,205 @@
...
+
+Once you have successfully flashed the firmware to the board, you may go to the next step, where you will write, compile, and flash a Buzz script.
+
+## Writing a Buzz Script
+
+ User-defined Buzz scripts are located within the [HiveMind](https://github.com/SwarmUS/HiveMind) codebase, in folder `bittybuzz/buzz_scripts`. 
+ 
+ ![Location of the Buzz scripts](img/buzz-scripts-location.png)
```
[the remark this answers is not in the document]
Comment (stlaperriere) [reply docs/sections/user-guide/basic-examples/led-flash-buzz.md:33]: removed

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 6]
```diff
@@ -1,11 +1,205 @@
...
+
+### The Script: `blinky.bzz`
+
+The next step is to create a new file called `blinky.bzz` under `bittybuzz/buzz_scripts`.
+
+This file should provide a `create_exec()` function to be used within `main.bzz` as an entrypoint. 
+
+> This is not an absolute rule; users could write everything directly in the `main.bzz` file, but this implementation makes it easier to switch from one buzz script to the other by simply changing the included file in the main.
+
+The `create_exec()` function must return an `executor` object. An `executor` is simply an object which will call a function at regular intervals (see [`executor.bzz`](https://github.com/SwarmUS/HiveMind/blob/master/src/bittybuzz/buzz_scripts/utils/executor.bzz)).
```
[the remark this answers is not in the document]
Comment (stlaperriere) [reply docs/sections/user-guide/basic-examples/led-flash-buzz.md:68]: changed

File: docs/sections/user-guide/basic-examples/led-flash-buzz.md [hunk 7]
```diff
@@ -1,11 +1,205 @@
...
+
+ctx = {
+    .led = 0,
+}
+
+function tick(context){
+
+}
+
+function create_exec(){
```
[the remark this answers is not in the document]
Comment (stlaperriere) [reply docs/sections/user-guide/basic-examples/led-flash-buzz.md:84]: done

**Pull request merged**